lib/Bindings/Python/circt/dialects/_hw_ops_ext.py

```diff
@@ -146,17 +146,23 @@ def __init__(
 
     input_types = []
     input_names = []
+    input_locs = []
     for (i, (port_name, port_type)) in enumerate(input_ports):
       input_types.append(port_type)
       input_names.append(StringAttr.get(str(port_name)))
+      input_locs.append(Attribute.parse("loc(unknown)"))
```

Comment:
@teqdruid They python bindings had to be updated to attach these new attributes.  Right now they are defaulting to `unknown`.  If a block is attached to the HWModuleOp, the locations stored in the block arguments should match these attributes. If we want better location information, here is where we would add it.

Also as a side note, we needed a `LocationAttr` not a `Location` here, so we had to reach for `Attribute.parse`.

Reply:
It `LocationAttr` is missing from `mlir.ir`, it should be added there.

Reply:
Python bindings: https://reviews.llvm.org/D142522